docs/tutorials/field_type/7_add_content_and_edit_views.md

```diff
@@ -1,275 +1,116 @@
 
-```javascript
-//Resources/public/js/views/fields/ez-tweet-view.js
+``` php
+// Form/TweetValueTransformer.php
+namespace EzSystems\TweetFieldTypeBundle\Form;
 
-YUI.add('ez-tweet-view', function (Y) {
-    "use strict";
-    /**
-     * Provides the Tweet field view
-     *
-     * @module ez-tweet-view
-     */
-    Y.namespace('eZ');
+use Symfony\Component\Form\DataTransformerInterface;
+use eZ\Publish\API\Repository\FieldType;
+use eZ\Publish\Core\FieldType\Value;
 
+class TweetValueTransformer implements DataTransformerInterface
+{
     /**
-     * The Tweet field view
-     *
-     * @namespace eZ
-     * @class TweetView
-     * @constructor
-     * @extends eZ.FieldView
+     * @var FieldType
      */
-    Y.eZ.TweetView = Y.Base.create('tweetView', Y.eZ.FieldView, [], {
-        /**
-         * Returns the value to be used in the template. If the value is not
-         * filled, it returns undefined otherwise an object with a `url` entry.
-         *
-         * @method _getFieldValue
-         * @protected
-         * @return Object
-         */
-        _getFieldValue: function () {
-            var value = this.get('field').fieldValue, res;
+    private $fieldType;
 
-            if ( !value || !value.url ) {
-                return res;
-            }
-            res = {url: value.url};
+    public function __construct(FieldType $fieldType)
+    {
+        $this->fieldType = $fieldType;
+    }
 
-            return res;
+    public function transform($value)
+    {
+        if (!$value instanceof Value) {
+            return null;
         }
-    });
 
-    Y.eZ.FieldView.registerFieldView('eztweet', Y.eZ.TweetView);
-});
-```
+        return (string) $value;
+    }
 
-### Registering the view
+    public function reverseTransform($value)
+    {
+        if ($value === null) {
+            return $this->fieldType->getEmptyValue();
+        }
 
-We also need to add information about our freshly created files to the already created configuration. To achieve this, let's add the following snippet to the `yui.yml` file:
-```yaml
-                tweetview-ez-template:
-                    type: 'template'
-                    path: bundles/ezsystemstweetfieldtype/templates/fields/view/tweet.hbt
-                ez-tweet-view:
-                    requires: ['ez-fieldview', 'tweetview-ez-template']
-                    dependencyOf: ['ez-rawcontentview']
-                    path: bundles/ezsystemstweetfieldtype/js/views/fields/ez-tweet-view.js
+        return $this->fieldType->fromHash($value);
+    }
+}
 ```
 
-After these steps you should see correct information when viewing Content Item in "Content structure" tab in PlatformUI.
+Next point will be using above transformer in our FormMapper. This is doable with addModelTransformer() method. Remember to pass eZ\Publish\API\Repository\FieldType object as constructor.
+
+``` php
+// eZ/Publish/FieldType/Tweet/FormMapper.php
+(...)
```

Comment:
If someone makes a copy-paste, what the (...) will become in PHP code?
Maybe a comment with `//beginning of the class` or put these parentheses in a comment could be better here.

Reply:
Agreed, but this is something that crops up in other parts of the tutorial (e.g. https://doc.ezplatform.com/en/2.0/tutorials/field_type/5_implement_the_legacy_storage_engine_converter/#implementing-field-value-converters-tofieldvalue-and-tostoragevalue), so we'll have to fix that anyway in other pages when doing an audit.

Reply:
The example you gave is parenthesis (or brackets) but in a PHP comment, so it's copy-paste safe.